### PR TITLE
 Per workspace folders configuration support.

### PR DESCRIPTION
**Pull request by jerome-benoit**
Reference: https://github.com/amatiasq/vsc-sort-imports/issues/68

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "sort-imports",
   "displayName": "sort-imports",
   "description": "Sort ES6 imports automatically.",
-  "version": "6.1.0",
+  "version": "6.1.1",
```
Comment (jquense) [inline package.json:5]: don't bump the version yourself, the release process will do that

Comment (jerome-benoit) [reply package.json]: Done, thanks.

**Comment by jerome-benoit** [comment]
> LGTM

Do you think it will be merged soon? 
And when will a new version be pushed on the extensions directory? 

I've disabled the extension in git on all projects I'm working on because of that issue.

**Comment by amatiasq** [comment]
@jerome-benoit I'm all in to add you as a collaborator so you can test and merge it yourself but @jquense has last word here.

Once this is merged me or @jquense can release a new version.

**Comment by jerome-benoit** [comment]
> @jerome-benoit I'm all in to add you as a collaborator so you can test and merge it yourself but @jquense has last word here.
> 
> Following ["The Pull Request Hack"](https://felixge.de/2013/03/11/the-pull-request-hack.html) here.

Not sure I will have time to do enhancements but at least bug fixes, probably. 

> Once this is merged @jquense or I can release a new version.

That PR is well tested locally. I really think it's release material but feel free to cross test ;-)

**Comment by jerome-benoit** [comment]
Any chance to have it merged?

**Comment by amatiasq** [comment]
@jerome-benoit apologies, apparently Jason Quense is not around.

I've added, now you can test it and merge yourself.

Once that's done please ping me and I'll deploy it.

**Comment by jerome-benoit** [comment]
> @jerome-benoit apologies, apparently Jason Quense is not around.
> 
> I've added, now you can test it and merge yourself.

Thanks. 

> Once that's done please ping me and I'll deploy it.

Is there a kind of beta test phase in the code extensions deployment system?  

That PR is tested locally but it might be wise to have a beta testing phase ... or at least cross-testing.

**Comment by amatiasq** [comment]
> Is there a kind of beta test phase in the code extensions deployment system?
> 
> That PR is tested locally but it might be wise to have a beta testing phase ... or at least cross-testing.

That's a good idea, we don't have one in place and I wouldn't know how to do that.

**Comment by jerome-benoit** [comment]
> > Is there a kind of beta test phase in the code extensions deployment system?
> > That PR is tested locally but it might be wise to have a beta testing phase ... or at least cross-testing.
> 
> That's a good idea, we don't have one in place and I wouldn't know how to do that.

All I can say is that the code merged in master is working properly within workspace that have several folders with different sorting configurations. 

So if you want to make a new extension release without more testing , that's fine by me.

**Comment by amatiasq** [comment]
I'm not able to deploy it, `vsce` is not accepting my token for some reason, will try again this afternoon

**Comment by jerome-benoit** [comment]
> I'm not able to deploy it, `vsce` is not accepting my token for some reason, will try again this afternoon

I've not made the version bump in package.json and README.md. You have to do it before releasing.

**Comment by jerome-benoit** [comment]
> I'm not able to deploy it, `vsce` is not accepting my token for some reason, will try again this afternoon

ping? 

I think I need another way to contact you than comments on GitHub to solve the issue. Is using your @gmail.com adresse fine?

**Comment by jquense** [comment]
(I also tried a publish but no dice from vsce)

**Comment by jerome-benoit** [comment]
> (I also tried a publish but no dice from vsce)
> 
> 

Well, I’m not familiar with the extension publication process but I’m pretty sure you can the tool with debug mode and have support from MS.

**Comment by jerome-benoit** [comment]
> 
>> (I also tried a publish but no dice from vsce)
>> 
>> 
> 
> Well, I’m not familiar with the extension publication process but I’m pretty sure you can the tool with debug mode and have support from MS.
> 


I've made a few cleanups and a fix in the master branch for packaging and publishing the extension with vsce:

npm run vsce:package -> build the extension and instal it locally on code;
npm run vsce:publish -> should publish it (untested)

Do you think one of you will be able to push the 6.1.1 version in a timely manner? And if it’s not working properly report the full error message here?

Thanks.

**Comment by amatiasq** [comment]
I don't know what you did but it worked. 6.2.0 deployed.

**Comment by jerome-benoit** [comment]
> 
> I don't know what you did but it worked. 6.2.0 deployed.
> 
> 
> 

Thanks.

I thought the version will be 6.1.1.

I’ve synced the changelog version in the master branch. Is it possible to republish the same version?

You probably have to unpublish it first and then publish it again by forcing the semver:

https://code.visualstudio.com/api/working-with-extensions/publishing-extension <https://code.visualstudio.com/api/working-with-extensions/publishing-extension>

++

**Comment by amatiasq** [comment]
This is not a bugfix, it's a new feature so it's a minor version upgrade.

I don't know what you need now but I could barely found the time to publish it, I have personal issues that make the time I can dedicate to this (or any) project very scarse. In fact I'm supposedly [not in charge of it anymore](https://github.com/amatiasq/vsc-sort-imports/issues/41).

If you're dedicated to see more deployments I can add you to the publisher so you can deploy yourself. And yes, you can email me at [my github username] at gmail.com

**Pull request merged**